chainer/optimizers/adam.py

```diff
@@ -49,6 +49,13 @@ def _learning_rate(hp, t):
         raise RuntimeError(
             'Can\'t determine the learning rate of Adam optimizer '
             'because the update steps have not been started.')
+
+    '''
+    In the paper of Adam, the 2nd-moment estimate v_t is
+    corrected before "+eps".
+    Chainer's Adam corrects 2nd-moment after "+eps".
+    It shouldn't cause any significant difference.
+    '''
```

Comment:
The change about "+eps" is suggested *in the Adam paper* and the `eps` argument of `optimizers.Adam` corresponds to $\hat{eps}$ there.

> Note that the efficiency of algorithm 1 can, at the expense of clarity, be improved upon by changingthe order of computation, e.g.  by replacing the last three lines in the loop with the following lines: αt=α·√1−βt2/(1−βt1) and θt←θt−1−αt·mt/(√vt+ ˆε).

https://arxiv.org/abs/1412.6980